src/pages/Dictionary/components/EditMode.jsx

```diff
@@ -1,53 +1,104 @@
-import { Button, Textarea } from '@nextui-org/react';
+import { useRef } from 'react';
+import { Button, Textarea, useDisclosure } from '@nextui-org/react';
+import { toast } from 'sonner';
+import { useNavigate } from 'react-router-dom';
+import getEditEntriesMutation from '../../../utils/getEditEntriesMutation';
+import EditConfirmModal from './EditConfirmModel';
 
 const EditMode = ({ entry, setEditEntryid }) => {
+  const { isOpen, onOpen, onOpenChange } = useDisclosure();
+  const definition = useRef(entry.definition);
+  const example = useRef(entry.example);
+
+  const navigate = useNavigate();
+  const mutation = getEditEntriesMutation(entry.id);
+
   const handleEditCancel = () => {
     setEditEntryid(undefined);
   };
 
-  const handleEditSubmit = (e) => {
+  const handleUpdate = (e) => {
     e.preventDefault();
-    const formData = new FormData(e.target);
-    const definition = formData.get('definition');
-    const example = formData.get('example');
+    onOpen();
+  };
+
+  const handleEditSubmit = () => {
+    const definitionTrim = definition.current.value.trim();
+    const exampleTrim = example.current.value.trim();
 
-    // TODO: handle edit confirm
-    console.log(definition, example);
+    if (definitionTrim === '' || exampleTrim === '') {
+      toast.error('Missing required input...');
+    } else {
+      mutation.mutate(
+        {
+          ...entry,
+          definition: definitionTrim,
+          example: exampleTrim
+        },
```

Comment:
you can have {merge: true} option and then this can be just
```
{
  definition: definitionTrim,
  example: exampleTrim
}
```

Reply:
im sorry the merge option is not here

Reply:
it should be included here 'useFirestoreDocumentMutation(entryDocRef, { merge: true });'

Reply:
Yes, I added it.